lib/ingreedy/amount_parser.rb

```diff
@@ -59,7 +59,7 @@ class AmountParser < Parslet::Parser
     private
 
     def word_digits
-      Ingreedy.dictionaries.current.numbers.keys
+      Ingreedy.dictionaries.current.numbers.keys.sort_by(&:length).reverse
```

Comment:
I could be wrong, but I remember seeing `Ingreedy.dictionaries.current.numbers` somewhere else in the gem, maybe it needs the same sorting?

---

Also, I looked for a way to do a reverse sort without having to go through the array twice, and didn't find any method... :( But I found [this StackOverflow answer](https://stackoverflow.com/a/2651028/2341409) that suggests that `.sort_by(&:length).reverse!` might be faster.

```rb
ary = []
10000.times { ary << rand(10000) }
n = 500
Benchmark.bm(20) do |x|
  x.report("sort.reverse")       { n.times { ary.sort.reverse } }
  x.report("sort.reverse!")   { n.times { ary.sort.reverse! } }
end
```